### PR TITLE
custom-editors: Fix the issue: Auto Save on Binary Custom Editor does not work correctly

### PR DESCRIPTION
**Pull request by safisa**
#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/11460.
Fix an issue where auto save on custom editor does not work correctly.

#### How to test

1. enable the auto-save
2. Use the vscode cats example and open the binary file example.pawDraw
3. do a change, and make sure that the file was saved
4. make sure that the save is not called recursively, you can see it by adding a breakpoint in the custom-editors-main.ts file using dev-tools, on the method change of the MainCustomEditorModel class.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Safi Seid-Ahmad <safi@k2view.com>

### CODE REVIEW

**Comment by safisa** [comment]
Hi @vince-fugnitto  @msujew 
Can you review this PR?

**Comment by vince-fugnitto** [comment]
> Hi @vince-fugnitto @msujew Can you review this PR?

@safisa do you mind sharing the `custom-editor-sample` you used, I see the following locally:

Screen Shot 2022-08-24 at 1 42 05 PM

**Comment by safisa** [comment]
Hi @vince-fugnitto

First, thank you for the review. 
Actually, I am using a custom editor that I can't share here, since it requires many other dependencies. I also checked the vscode cats pawDraw example (this uses the binary custom editor), and there is an issue in the example since it uses API that is not supported in Theia (vscode.workspace.fs.isWritableFileSystem - there is already an opened issue on this).
I did some changes in the example by removing the isWritableFileSystem API, but still had an issue in loading the image in the pawDraw.js file...
Anyway, now you can use the attached updated example. When opening the example.pawDraw file, you will get a blank canvas with no image (image load issue!), and then start drawing, you will see that the file becomes dirty, and on save you will see that the file on disk is updated (check file timestamp or size), but when reopened you will start from blank again because of the image load issue.

If you enable the auto-save, and open and edit the file as above, before this PR, the save event will keep being invoked infinitely. you can check it by testing the timestamp of the file on disk, or putting a breakpoint in the pawDraw.js file inside the message event (getFileData) or in the Theia custom-editors-main.ts file, on the method change of the MainCustomEditorModel class.
After this PR fix the auto-save works as expected.
I need this fix since we are suing many custom editors in our App.

Thanks in advance.

[cat-customs-0.0.1.vsix.zip](https://github.com/eclipse-theia/theia/files/9423258/cat-customs-0.0.1.vsix.zip)

**Comment by vince-fugnitto** [comment]
@safisa can you please fix the git history (seems to be issues after rebasing) and ping me when it's ready for another review?

**Comment by safisa** [comment]
Hi @vince-fugnitto
Fixed the git history, now there is only the 2 files I have changed

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@
 - [vsx-registry] improved styling of the `Extensions` view [#11494](https://github.com/eclipse-theia/theia/pull/11494)
 - [vsx-registry] removed localization for `Open VSX Registry` [#11523](https://github.com/eclipse-theia/theia/pull/11523)
 - [vsx-registry] updated extension editor rendering [#11605](https://github.com/eclipse-theia/theia/pull/11605)
+- [custom-editor] Auto Save on Binary Custom Editor does not work correctly [#11460](https://github.com/eclipse-theia/theia/issues/11460)
```
Comment (vince-fugnitto) [inline CHANGELOG.md:45]: minor: `1.29.0` was released last week, we'd need to add it under a new `1.30.0` entry. 
You can remove it for now, I generally update the changelog prior to the release anyways.

**Review by vince-fugnitto — APPROVED** [review]
I confirmed the bug on master and that it is correctly fixed by the updates 👍

**Comment by safisa** [comment]
Hi @vince-fugnitto 
This PR is already approved, is it going to be merged before the next Theia version?

**Pull request merged**